optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/impl/ConsecutiveIntervalInfoImpl.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.common.experimental.impl;
+
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.NavigableSet;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.function.BiFunction;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.optaplanner.examples.common.experimental.api.ConsecutiveIntervalInfo;
+import org.optaplanner.examples.common.experimental.api.IntervalBreak;
+import org.optaplanner.examples.common.experimental.api.IntervalCluster;
+
+public class ConsecutiveIntervalInfoImpl<IntervalType_, PointType_ extends Comparable<PointType_>, DifferenceType_ extends Comparable<DifferenceType_>>
+        implements
+        ConsecutiveIntervalInfo<IntervalType_, PointType_, DifferenceType_> {
+    private final NavigableMap<IntervalSplitPoint<IntervalType_, PointType_>, IntervalClusterImpl<IntervalType_, PointType_, DifferenceType_>> clusterStartSplitPointToCluster;
```

Comment:
The generics are definitely getting to an insane level here. That said, I don't think there's much we can do. Perhaps we remove the `Type_` suffixes, because it is obvious that they're going to be types?